sdk/src/main/java/com/horizen/cryptolibprovider/CswCircuitImplZendoo.java

```diff
@@ -22,6 +27,32 @@ public FieldElement getUtxoMerkleTreeLeaf(Box<Proposition> box) {
         return utxo.getNullifier();
     }
 
+    @Override
+    public byte[] transformPrivateKey25519(PrivateKey25519 pk) {
```

Comment:
The function that performs this transformation seems to be public in Bouncy Castle:

https://github.com/bcgit/bc-java/blob/1ef1da015e116c1c2d09719674dfdd437971077f/core/src/main/java/org/bouncycastle/math/ec/rfc8032/Ed25519.java#L334

Called by:

https://github.com/bcgit/bc-java/blob/1ef1da015e116c1c2d09719674dfdd437971077f/core/src/main/java/org/bouncycastle/crypto/params/Ed25519PrivateKeyParameters.java#L61

Perhaps you can use directly that one instead ?

Reply:
we can't use Bouncy Castle directly, because the methods are private.
I've added tests against original solution using reflection.